src/components/TeacherComponents/LessonList/LessonList.tsx

```diff
@@ -48,6 +50,10 @@ const LessonList: FC<LessonListProps> = (props) => {
     onClick,
   } = props;
   const { currentPage, pageSize, firstItemRef } = paginationProps;
+
+  const router = useRouter();
+
+  const paginationRoute = router.asPath.split("?")[0] || router.asPath;
```

Comment:
could we return this and the onPageChange fn from pagination props?

Reply:
Moved both into usePagination hook ✅